root_pandas/readwrite.py

```diff
@@ -147,6 +148,18 @@ def do_flatten(arr):
         arr = append_fields(arr_, '__array_index', idx, usemask=False, asrecarray=True)
         return arr
 
+    def remove_high_dimensions(arr):
+        allowed_dimensions = [0]
+        first_row = arr[0]
+        good_cols = [True if x.ndim in allowed_dimensions else False for x in first_row]
+        col_names = np.array(list(arr.dtype.names))
```

Comment:
Why do you need the `list` here?
Looks like `arr.dtype.names` is a tuple, which we should be able to convert to an array directly.